File: C++/anagram.cc
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        int count[26] = {0};

        //Count freq of chars in s
        for (char c: s){
            count[c-'a']++;
        }
        //Decrement freq of chars in t
        for (char c: t){
            count[c-'a']--;
        }
        //if any chars has non-zero frequency => not anagram
        for (int v: count){
            if (v != 0) return false;
        }
        return true;
    }
};

File: Python/leetcode/anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        count = [0] * 26

        # count freq of chars in s to count array
        for c in s:
            count[ord(c) - ord('a')] +=1
        #decrement freq of chars in t to count array
        for c in t:
            count[ord(c) - ord('a')] -=1
        #check if any chars in count array has non-0 freq => yes = not anagram
        for v in count:
            if v != 0:
                return False

        return True